kubernetes/utils/deserialize.py

```diff
@@ -0,0 +1,285 @@
+                    "response_type: {} lookup info: {}".format(
+                        fnc_lookup,
+                        k8s_api,
+                        response_type,
+                        info))
+    else:
+        msg = "Failed to find a function to inspect; lookup info: {}".format(
+            info)
+        raise FailToLoadError(reason=msg)
+
+    return response_type
+
+
+def _find_response_type(func):
+    for line in pydoc.getdoc(func).splitlines():
+        if line.startswith(PYDOC_RETURN_LABEL):
+            return line[len(PYDOC_RETURN_LABEL):].strip()
+    return ""
+
+
+def load_from_dict(data, klass=None, verbose=False):
+    """
+    Load object/objects from a dictionary containing valid kubernetes
+    API object (i.e. List, Service, etc).
+
+    Input:
+    data: a dictionary holding valid kubernetes objects
+    klass: class literal for deserialized object,
+    or string of class name.
+    verbose: If True, print additional info.
+        Default is False.
+    Raises:
+        FailToLoadError which holds list of load failures.
+    """
+    # If it is a list type, will need to iterate its items
+    load_exceptions = []
+    obj = None
+    if "List" in data["kind"]:
+        obj_list = []
+        # Could be "List" or "Pod/Service/...List"
+        # This is a list type. iterate within its items
+        kind = data["kind"].replace("List", "")
+        if verbose:
+            print("List detected # of items: {}".format(len(data["items"])))
+        for item in data["items"]:
+            # Mitigate cases when server returns a xxxList object
+            # See kubernetes-client/python#586
+            if kind is not "":
+                item["apiVersion"] = data["apiVersion"]
+                item["kind"] = kind
+            try:
+                obj = load_single_item(item, klass=klass, verbose=verbose)
+                obj_list.append(obj)
+            except FailToLoadError as load_exception:
+                load_exceptions.append(load_exception)
+
+        obj = obj_list
+
+    else:
+        try:
+            obj = load_single_item(data, klass=klass, verbose=verbose)
+        except FailToLoadError as load_exception:
+            load_exceptions.append(load_exception)
+
+    # In case we have exceptions waiting for us, raise them
+    if load_exceptions:
+        raise FailToLoadError(load_exceptions)
+
+    return obj
+
+
+def load_single_item(data, klass=None, verbose=False):
+    """
+    Load a single object from a dictionary containing valid kubernetes
+    API object (i.e. List, Service, etc).
+    Input:
+    data: a dictionary holding a valid kubernetes object
+    klass: class literal for deserialized object,
+    or string of class name.
+    verbose: If True, print additional info.
+        Default is False.
+    """
+    # Mock api response from dict
+    resp_mock = RespMock()
+    resp_mock.data = json.dumps(data)
+
+    # Infer response_type from json data when not provided
+    if not klass:
```

Comment:
I think having the caller to pass in the klass is the simplest and satisfies most use cases.

Reply:
I'll trying to target both use-cases. The one we pass klass and the auto-inferring. I will prioritize the first one. For auto-inferring I am taking inspiration from the watch implementation.

Reply:
I have added the klass parameter as an optional one; if one is not given it will try to find the proper klass with the help of the response_type_from_dict function.